### PR TITLE
DKC Config update version 1

### PR DESCRIPTION
**Pull request by BrentEaston**

### CODE REVIEW

File: vassal-app/src/main/java/VASSAL/configure/ConfigurableList.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ */
+public interface ConfigurableList {
+  void moveEntryUp();
+  void moveEntryDown();
+  void moveEntryTop();
+  void moveEntryBottom();
+  void addEntry();
+  void deleteEntry(ConfigurableListEntry entry);
+  void editEntry();
+  JPanel getListController();
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/configure/ConfigurableList.java:33]: Could this be a JComponent? If nothing depends on it being a JPanel, that would make the interface more flexible.

File: vassal-app/src/main/java/VASSAL/build/module/properties/PropertyChangerConfigurer.java [hunk 2]
```diff
@@ -303,4 +314,34 @@ public void setValue(String s) {
...
+  @Override
+  public void setHighlighted(boolean highlighted) {
+    super.setHighlighted(highlighted);
+    typeConfig.setHighlighted(highlighted);
+    incrConfig.setHighlighted(highlighted);
+    valueConfig.setHighlighted(highlighted);
+    promptConfig.setHighlighted(highlighted);
+    validValuesConfig.setHighlighted(highlighted);
+  }
+
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/build/module/properties/PropertyChangerConfigurer.java:327]: Could you do something like
```
final List<Configurer> configurers = List.of(typeConfig, incrConfig, valueConfig, promptConfig, validValuesConfig);

@Override
public void setHighlighted(boolean highlighted) {
  super.setHighlighted(highlighted);
  configurers.forEach(c -> c.setHighlighted(highlighted));
}
```
and similarly for the two functions following?

File: vassal-app/src/main/java/VASSAL/configure/DynamicKeyCommandListConfigurer.java [hunk 3]
```diff
@@ -0,0 +1,578 @@
...
+        c.setValue(sd.nextToken());
+        getListValue().add(c.getValue());
+      }
+    }
+    rebuildControls();
+  }
+
+  protected void updateValue() {
+    noUpdate = true;
+    final ArrayList<Object> newArray = new ArrayList<>();
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/configure/DynamicKeyCommandListConfigurer.java:111]: `final List<Object>`?

File: vassal-app/src/main/java/VASSAL/configure/ConfigurableList.java (hunk 1 above)
Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/configure/ConfigurableList.java:33]: True. Done.

File: vassal-app/src/main/java/VASSAL/build/module/properties/PropertyChangerConfigurer.java (hunk 2 above)
Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/build/module/properties/PropertyChangerConfigurer.java:327]: Done

File: vassal-app/src/main/java/VASSAL/configure/DynamicKeyCommandListConfigurer.java (hunk 3 above)
Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/configure/DynamicKeyCommandListConfigurer.java:111]: Fixed

**Comment by BrentEaston** [comment]
I've got a few more changes coming through on this one to make it more generic for use in other list Configurers. Have a pesky problem with a Test failing, where the actual code does not.

**Comment by BrentEaston** [comment]
need to recode

**Pull request closed without merging**